Source/WebCore/crypto/mac/CryptoKeyECMac.cpp

```diff
@@ -90,48 +96,94 @@ static constexpr bool doesFieldElementMatchNamedCurve(CryptoKeyEC::NamedCurve cu
+            return std::nullopt;
+        auto publicKey = CryptoKeyEC::create(identifier, curve, CryptoKeyType::Public, std::unique_ptr<typename std::remove_pointer<CCECCryptorRef>::type, WebCore::CCECCryptorRefDeleter>(ccPublicKey), true, usages);
+        auto privateKey = CryptoKeyEC::create(identifier, curve, CryptoKeyType::Private, std::unique_ptr<typename std::remove_pointer<CCECCryptorRef>::type, WebCore::CCECCryptorRefDeleter>(ccPrivateKey), extractable, usages);
+        return CryptoKeyPair { WTFMove(publicKey), WTFMove(privateKey) };
+        
+    }
+    ccec25519pubkey ccPublicKey;
+    ccec25519secretkey ccPrivateKey;
+
+    int ccerror = 0;
+    const struct ccdigest_info* di = ccsha512_di();
+    struct ccrng_state* rng = ccrng(&ccerror);
 
-    auto publicKey = CryptoKeyEC::create(identifier, curve, CryptoKeyType::Public, PlatformECKeyContainer(ccPublicKey), true, usages);
-    auto privateKey = CryptoKeyEC::create(identifier, curve, CryptoKeyType::Private, PlatformECKeyContainer(ccPrivateKey), extractable, usages);
+    cced25519_make_key_pair(di, rng, ccPublicKey, ccPrivateKey);
+    auto publicKey = CryptoKeyEC::create(identifier, curve, CryptoKeyType::Public, Vector<uint8_t>(ccPublicKey), true, usages);
+    auto privateKey = CryptoKeyEC::create(identifier, curve, CryptoKeyType::Private, Vector<uint8_t>(ccPrivateKey), extractable, usages);
     return CryptoKeyPair { WTFMove(publicKey), WTFMove(privateKey) };
+    
 }
 
 RefPtr<CryptoKeyEC> CryptoKeyEC::platformImportRaw(CryptoAlgorithmIdentifier identifier, NamedCurve curve, Vector<uint8_t>&& keyData, bool extractable, CryptoKeyUsageBitmap usages)
 {
-    if (!doesUncompressedPointMatchNamedCurve(curve, keyData.size()))
-        return nullptr;
-
-    CCECCryptorRef ccPublicKey = nullptr;
-    if (CCECCryptorImportKey(kCCImportKeyBinary, keyData.data(), keyData.size(), ccECKeyPublic, &ccPublicKey))
+    if (curve != NamedCurve::Curve25519) {
+        if (!doesUncompressedPointMatchNamedCurve(curve, keyData.size()))
+            return nullptr;
+        CCECCryptorRef ccPublicKey = nullptr;
+        if (CCECCryptorImportKey(kCCImportKeyBinary, keyData.data(), keyData.size(), ccECKeyPublic, &ccPublicKey))
+            return nullptr;
+        return create(identifier, curve, CryptoKeyType::Public, std::unique_ptr<typename std::remove_pointer<CCECCryptorRef>::type, WebCore::CCECCryptorRefDeleter>(ccPublicKey), extractable, usages);
+        
+    }
+    if (!doesFieldElementMatchNamedCurve(curve, keyData.size()))
         return nullptr;
-
-    return create(identifier, curve, CryptoKeyType::Public, PlatformECKeyContainer(ccPublicKey), extractable, usages);
+    if (usages & CryptoKeyUsageSign)
+        return create(identifier, curve, CryptoKeyType::Private, Vector<uint8_t>(keyData), extractable, usages);
+    return create(identifier, curve, CryptoKeyType::Public, Vector<uint8_t>(keyData), extractable, usages);
```

Comment:
Seems this could be written like this:
```
auto keyType = usages & CryptoKeyUsageSign ? CryptoKeyType::Private : CryptoKeyType::Public;
return create(identifier, curve, keyType, Vector<uint8_t>(keyData), extractable, usages);
```
to avoid duplication.

Reply:
Chris comment does not seem addressed.

Reply:
changed